### PR TITLE
Add examples of coxeter usage

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Add some examples of using coxeter's features in the context of examples.

## Motivation and Context

Suggested as part of the paper review process on openjournals/joss-reviews#3098.

## Types of Changes

- [x] Documentation update
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/coxeter/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/coxeter/blob/master/ContributorAgreement.md).
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] I have updated the [changelog](https://github.com/glotzerlab/coxeter/blob/master/ChangeLog.txt) and the [credits](https://github.com/glotzerlab/coxeter/blob/master/doc/source/credits.rst).

### CODE REVIEW

**Comment by wolfv** [comment]
wow, this looks great!

**Review by Tobias-Dwyer — APPROVED** [review]
The examples look correctly written to me (though, I have not run them in a jupyter notebook myself). The justifications added for the examples was nice as well and definitely strengthen the docs.

File: doc/readthedocs-env.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  - numpy
+  - rowan
+  - scipy
+  - sphinx
+  - sphinx_rtd_theme
+  - nbsphinx
+  - sphinxcontrib-bibtex>=2.0.0
+  - autodocsumm
+  - fresnel
+  - nb_conda_kernels
```
Comment (wolfv) [inline doc/readthedocs-env.yml:17]: i think if you install `jupyter` or `notebook` it should come with the appropriate python3 kernel. I believe `nb_conda_kernels` shouldn't be needed.

Comment (vyasr) [reply doc/readthedocs-env.yml]: Nice catch, thanks! I completely missed that `jupyter` wasn't in there.

**Comment by vyasr** [comment]
@wolfv After going the usual few rounds fighting with RTD I've got builds working, you can see them [here](https://coxeter--151.org.readthedocs.build/en/151/examples.html).

**Comment by wolfv** [comment]
great examples, I think they really shine! I just checked them out under https://coxeter--151.org.readthedocs.build/en/151/examples.html

**Review by Tobias-Dwyer — APPROVED** [review]

**Pull request merged**